src/common/transformations/src/transformations/common_optimizations/adaptive_pool_to_reduce.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/adaptive_pool_to_reduce.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::AdaptivePoolToReduce::AdaptivePoolToReduce() {
+    MATCHER_SCOPE(AdaptivePoolToReduce);
+    auto data_pattern = pattern::any_input();
+    auto out_spatial_shape = pattern::wrap_type<opset11::Constant>();
+    auto a_avg_pool = pattern::wrap_type<opset11::AdaptiveAvgPool>({data_pattern, out_spatial_shape});
```

Comment:
`auto a_pool = pattern::wrap_type<opset11::AdaptiveAvgPool, opset11::AdaptiveMaxPool>({data_pattern, out_spatial_shape}); `instead of Or pattern